### PR TITLE
Add following_tags_count to user sidebar stats cache key

### LINKED ISSUE #5657: Following a tag should update user sidebar stats
**Describe the bug**

After following a tag, the number of "followed" tags statistic on the user's sidebar does not update accordingly.

**To Reproduce**

1. Check how many tags you follow on your profile.
2. Go follow another tag.
3. Check your profile again, looking at the sidebar; the number won't have updated.

**Expected behavior**

I'd expect the "followed tags" count to increment.

**Screenshots**
On the dashboard, the number of tags is correctly accounted for:
Screen Shot 2020-01-22 at 11 24 38 AM

However, on the user page, the side bar stats don't reflect this:
Screen Shot 2020-01-22 at 11 24 13 AM

**Desktop (please complete the following information):**

OS: Mac Mojave 10.14.5
Browser: Chrome
Version: 79.0.3945.130 (Official Build) (64-bit)

**Additional context**

I suspect this is related to the caching logic behind tags; would probably start by looking into `@user.decorate.cached_followed_tags`.

### PR DESCRIPTION
**Pull request by krzysiek1507**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adding `following_tags_count` gets rid of the issue when `@user.last_followed_at` doesn't change.
This line is quite long. Is there any "repo" for cache keys?

## Related Tickets & Documents

Closes #5657

## QA Instructions, Screenshots, Recordings

Play with (un)following tags and check if the counter on the user profile changes.

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

File: app/views/users/_sidebar.html.erb [hunk 1]
```diff
@@ -44,7 +44,7 @@
           </div>
         <% end %>
       <% end %>
-      <% cache "user-profile-sidebar-stats-#{@user.id}-#{@user.last_article_at}-#{@user.last_comment_at}-#{@user.last_followed_at}", expires_in: 10.days do %>
+      <% cache "user-profile-sidebar-stats-#{@user.id}-#{@user.last_article_at}-#{@user.last_comment_at}-#{@user.following_tags_count}-#{@user.last_followed_at}", expires_in: 10.days do %>
```
Comment (mstruve) [inline app/views/users/_sidebar.html.erb:47]:
```suggestion
      <% cache "user-profile-sidebar-stats-#{@user.id}-#{@user.last_article_at&.rfc3339}-#{@user.last_comment_at&.rfc3339}-#{@user.following_tags_count}-#{@user.last_followed_at&.rfc3339}", expires_in: 10.days do %>
```

While we are here could we format these timestamps a bit nicer for the cache key? Using rfc3339 removes all of the spaces which can make searching for the keys easier. 
```ruby
[41] pry(main)> User.last.created_at.rfc3339
=> "2020-10-19T18:13:01Z"
[43] pry(main)> User.last.created_at.to_s
=> "2020-10-19 18:13:01 UTC"
```

**Review by mstruve — APPROVED** [review]
Thanks for the fix! The code looks good. Offered a small suggestion to improve the readability of the cache key, feel free to add it if you would like.

**Comment by krzysiek1507** [comment]
Thanks @mstruve. I applied your suggestion!

**Review by vaidehijoshi — APPROVED** [review]
This works beautifully -- thank you for taking the time to fix this, @krzysiek1507!! 💯

**Pull request merged**